### PR TITLE
[MM-35] Update plugin with respect to phase 1 upgrades

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Updated plugin with respect to phase 1 upgrades

- Updated Makefile
- Updated version in `nvmrc` file and update `package-lock.json` file.
- Updated developer docs link in the readme file

### CODE REVIEW

File: webapp/package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
         "fix": "eslint --ignore-pattern node_modules --ignore-pattern dist --ext .js --ext .jsx . --quiet --fix",
         "test": "jest --forceExit --detectOpenHandles --verbose",
         "test:watch": "jest --watch",
-        "test-ci": "jest --forceExit --detectOpenHandles --maxWorkers=2"
+        "test-ci": "jest --forceExit --detectOpenHandles --maxWorkers=2",
```
Comment (mickmister) [inline webapp/package.json:12]: Note that we'll also want to sync with the starter template the scripts in `package.json` as well. I can close https://github.com/mattermost/mattermost-plugin-gitlab/pull/421 in favor of this PR

File: Makefile [hunk 2]
```diff
@@ -262,15 +258,6 @@ ifneq ($(HAS_WEBAPP),)
 endif
 	rm -fr build/bin/
 
-## Sync directory with a starter template
-sync:
```
Comment (mickmister) [inline Makefile:266]: We should also remove the `build/sync` folder that this interacts with

**Review by mickmister — COMMENTED** [review]
Thanks for these updates @ayusht2810 :+1: I have two remaining requests on this. I've updated the doc to reflect this as well https://docs.google.com/document/d/12IoQTHxFHc7yvM-oxPrBTx_LWW6lDVVego8TWkBOQE0/edit

**Comment by ayusht2810** [comment]
@mickmister fixed the review comments. Please re-review.

File: Makefile [hunk 3]
```diff
@@ -40,6 +46,7 @@ all: check-style test dist
 check-style: webapp/node_modules
 	@echo Checking for style guide compliance
 
+## TODO: Add check-types as well
```
Comment (mickmister) [inline Makefile:49]: What's the result when we run `npm run check-types` right now? Maybe the errors we can take care of while this is open? That way the project is ready to be worked on with other typescript contributions. No worries if you think it doesn't make sense to do that in this PR

**Review by mickmister — APPROVED** [review]
Awesome LGTM :+1:

File: Makefile (hunk 3 above)
Comment (ayusht2810) [reply Makefile:49]: @mickmister On adding the above script, we were getting various errors. So I think we can do that in another PR.

**Review by AayushChaudhary0001 — APPROVED** [review]
Tested for all the basic features and functionalities, working fine for all of them. Nothing seems to break, approved.

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by mickmister** [comment]
@ayusht2810 Heads up that there are linting issues to resolve here

Note that for the `no-use-before-define` rule, we can just disable it in `webapp/.eslintrc.json`, since the `@typescript-eslint/no-use-before-define` rule already takes care of this. This is a known issue with this eslint rule working with typescript. To disable the rule you can just set this value:

https://github.com/mattermost/mattermost-plugin-gitlab/blob/ee780c1c1b0497d641439ad3c3a8de1fb5f2ab2f/webapp/.eslintrc.json#L380-L387

to

```
"no-use-before-define": "off"
```

**Comment by ayusht2810** [comment]
@mickmister fixed the lint errors and typescript errors.

> I think we can modify the camelcase rule with ignoreDestructuring: true
Regarding the camelcase issue in the plugin, I tried to fix it by applying the above rule, but I was still getting camelcase lint errors in the plugin. For now, I have disabled the rule. Let me know your thoughts on this issue.

**Comment by mickmister** [comment]
@ayusht2810 Disabling the `camelCase` rule sounds fine to me. Thanks for looking into that

**Comment by mickmister** [comment]
Are we able to include the latest PRs from the starter template like https://github.com/mattermost/mattermost-plugin-starter-template/pull/192, or would you prefer to do them in a separate PR? A separate PR probably makes sense to minimize scope and get these merged.

I wonder if we can have an automated process to cherry-pick starter template PRs into plugin projects. That would make keeping things up-to-date a lot easier. We could batch them with multiple cherry-picks in one PR if we wanted to as well.

**Review by mickmister — APPROVED** [review]
LGTM, great work @ayusht2810  :rocket:

**Pull request merged**